pkg/graphql/resolver/utils.go

```diff
@@ -11,20 +11,31 @@ const HeadersKey string = "headers"
 
 func contextToHeaders(ctx context.Context, request *http.Request) {
 	if ctx.Value(HeadersKey) != nil {
-		headers, ok := ctx.Value(HeadersKey).(map[string]string)
+		headers, ok := ctx.Value(HeadersKey).(map[string]interface{})
 		if ok {
 			for key, value := range headers {
-				if value != "" {
-					request.Header.Add(key, value)
+				switch v := value.(type) {
+				case string:
+					if v != "" {
+						request.Header.Add(key, v)
+					}
+				case []string:
+					// Handle multiple values for the same header (e.g., Impersonate-Group)
+					for _, val := range v {
+						if val != "" {
+							request.Header.Add(key, val)
+						}
+					}
 				}
 			}
 		}
 	}
 }
 
 type initPayload struct {
-	ImpersonateUser  string `json:"Impersonate-User"`
-	ImpersonateGroup string `json:"Impersonate-Group"`
+	ImpersonateUser   string   `json:"Impersonate-User"`
+	ImpersonateGroup  string   `json:"Impersonate-Group"`
+	ImpersonateGroups []string `json:"Impersonate-Groups"`
```

Comment:
Im not really sure why we cant have just: `ImpersonateGroups []string` or `ImpersonateGroups []string` ? since its not an API